src/prefect/engine/task_runner.py

```diff
@@ -613,6 +616,39 @@ def load_results(
         """
         return state, upstream_states
 
+    def check_target(self, state: State, inputs: Dict[str, Result]) -> State:
+        """
+        Checks if a Result exists at the task's target.
+
+        Args:
+            - state (State): the current state of this task
+            - inputs (Dict[str, Result]): a dictionary of inputs whose keys correspond
+                to the task's `run()` arguments.
+
+        Returns:
+            - State: the state of the task after running the check
+        """
+        result = self.task.result
+        target = self.task.target
+
+        if result and target:
+            if result.exists(target):
```

Comment:
I think we need to update the signature of `result.exists` to accept `**kwargs` (just like `result.format`) which are used to optionally template the `target` location.  That way targets like `"data/{task_name}/{today}"` work

Reply:
For a target of `"data/{task_name}/{today}"` are `task_name` and `today` default values?

Reply:
Yup both of those are values we place in context